tests/beta/interfaces/test_qnode_autograd.py

```diff
@@ -0,0 +1,608 @@
+                [np.cos(y) * np.sin(x) / 2, np.cos(x) * np.sin(y) / 2],
+            ]
+        )
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_sampling(self, dev_name, diff_method):
+        """Test sampling works as expected"""
+        dev = qml.device(dev_name, wires=2, shots=10)
+
+        @qnode(dev, diff_method=diff_method, interface="autograd")
+        def circuit():
+            qml.Hadamard(wires=[0])
+            qml.CNOT(wires=[0, 1])
+            return [sample(qml.PauliZ(0)), sample(qml.PauliX(1))]
+
+        res = circuit()
+
+        assert res.shape == (2, 10)
+        assert isinstance(res, np.ndarray)
+
+
+def qtransform(qnode, a, interface=np):
+    """Transforms every RY(y) gate in a circuit to RX(-a*cos(y))"""
+
+    def construct(self, args, kwargs):
+        """New quantum tape construct method, that performs
+        the transform on the tape in a define-by-run manner"""
+
+        # the following global variable is defined simply for testing
+        # purposes, so that we can easily extract the transformed operations
+        # for verification.
+        global t_op
+
+        t_op = []
+
+        QNode.construct(self, args, kwargs)
+
+        new_ops = []
+        for o in self.qtape.operations:
+            # here, we loop through all tape operations, and make
+            # the transformation if a RY gate is encountered.
+            if isinstance(o, qml.RY):
+                t_op.append(qml.RX(-a * interface.cos(o.data[0]), wires=o.wires))
+                new_ops.append(t_op[-1])
+            else:
+                new_ops.append(o)
+
+        self.qtape._ops = new_ops
+        self.qtape._update()
+
+    import copy
+
+    new_qnode = copy.deepcopy(qnode)
+    new_qnode.construct = construct.__get__(new_qnode, QNode)
+    return new_qnode
+
+
+@pytest.mark.parametrize(
+    "dev_name,diff_method",
+    [("default.qubit", "finite-diff"), ("default.qubit.autograd", "backprop")],
+)
+def test_transform(dev_name, diff_method, monkeypatch, tol):
+    """Test an example transform"""
+    monkeypatch.setattr(qml.operation.Operation, "do_check_domain", False)
+
+    dev = qml.device(dev_name, wires=1)
+
+    @qnode(dev, interface="autograd", diff_method=diff_method)
+    def circuit(weights):
+        # the following global variables are defined simply for testing
+        # purposes, so that we can easily extract the operations for verification.
+        global op1, op2
+        op1 = qml.RY(weights[0], wires=0)
+        op2 = qml.RX(weights[1], wires=0)
+        return expval(qml.PauliZ(wires=0))
+
+    weights = np.array([0.32, 0.543], requires_grad=True)
+    a = np.array(0.5, requires_grad=True)
+
+    def loss(weights, a):
+        # the following global variable is defined simply for testing
+        # purposes, so that we can easily extract the transformed QNode
+        # for verification.
+        global new_circuit

```

Comment:
Could we have two empty lists just before the definition of `loss` (one for operations and one for parameters), append the operations and parameters of `new_circuit` in there and check the elements of that list later? Using `global` always makes me worried (especially when testing)